### ISSUE TITLE
Create a web-based GUI for teosd

### ISSUE THREAD

**Issue by sr-gi** [talaia-labs/rust-teos#192]
It would be nice to have a web-based GUI to monitor what is going on with `teosd`.

This should pull data from `teos-cli` and display it in a user-friendly manner.

Further info: https://www.summerofbitcoin.org/project-ideas-details?recordId=recYp3MeVaI79Jocg

DISCLAIMER
This is a Summer Of Bitcoin (SOB) project. If you're considering tackling this but you are not part of SOB, please leave it until SOB is over. After that, feel free to jump in.

**Comment by mariocynicys** [comment]
And would be so nice to have it integrated with [umbrel](umbrel.com).
Seems like a lot of lightning nodes are running on umbrel, they would probably appreciate having a one-click-install watchtower there.

Relevant repo: https://github.com/getumbrel/umbrel-apps

**Comment by sr-gi** [comment]
Oh, you already mentioned this here, I forgot 😅 . Anyway, I'm linking it so whoever takes this can also trey to accommodate that.

**Comment by mariocynicys** [comment]
> Oh, you already mentioned this here, I forgot sweat_smile . Anyway, I'm linking it so whoever takes this can also try to accommodate that.

No, the one I mentioned here I mean the tower with it, meaning that we can package the tower in a docker container and publish it along with it's to-be-made web interface into an umbrel app. Thus installing an altruistic tower on umbrel would be an easy single click.

The other one #207 is aimed for doing the same for the plugin. Integrating it with umbrel's CLN app.

**Comment by Devilinheaven234** [comment]
I'm a part of SOB and will be opting for this project would like to take it on

**Comment by sr-gi** [comment]
> I'm a part of SOB and will be opting for this project would like to take it on

I'd recommend you join slack and start working on the competency tests if you haven't already

**Comment by anipaul2** [comment]
> And would be so nice to have it integrated with [umbrel](umbrel.com). Seems like a lot of lightning nodes are running on umbrel, they would probably appreciate having a one-click-install watchtower there.
> 
> Relevant repo: https://github.com/getumbrel/umbrel-apps

I guess the link is broken. It doesn't takes to umbrel.com

**Comment by anipaul2** [comment]
> > Oh, you already mentioned this here, I forgot sweat_smile . Anyway, I'm linking it so whoever takes this can also try to accommodate that.
> 
> No, the one I mentioned here I mean the tower with it, meaning that we can package the tower in a docker container and publish it along with it's to-be-made web interface into an umbrel app. Thus installing an altruistic tower on umbrel would be an easy single click.
> 
> The other one #207 is aimed for doing the same for the plugin. Integrating it with umbrel's CLN app.

So, after making the Gui we can integrate with umbrel so that the tower can be run from there? And if it's done it can also be integrated with the cln plugin which is the watchtower client?

**Comment by jonmarrs** [comment]
I'd also recommend making an Umbrel app that is compatible with LND. There are two popular Lightning implementations that run on Umbrel: CLN and LND. We chould make apps for both versions of Lightning.

**Comment by sr-gi** [comment]
I still need to figure out how to properly integrate this within the repo, or the org, but @optimm did a great job creating a GUI for teosd: https://github.com/optimm/teosd-gui
